### ISSUE TITLE
Unexpected "Pattern will never be matched" warning/error after tuple case

### ISSUE THREAD

**Issue by mtomassoli** [microsoft/pyright#6670]
```python
def f(x: tuple[str, str] | str):
    match x:
        case (_, _):
            return 1
        case _:     # Pattern will never be matched for subject type "Never"
            return 2

assert f(('', '')) == 1
assert f('') == 2
```

Version 1.1.339

[Pyright Playground](https://pyright-play.net/?strict=true&code=CYUwZgBGAUAeBcEAuBXADgGxAbQM5ICcAaCfAgXQgB9TCBKeAWACgI2IBbAQyQGMALCAhbtREXl1wgI0APolZDEWJUEQqAgDsIARmUqJUiLMSiAxBAAKPJCC0QA7gEsMGCJpAA3OxABG07j5%2BEGAoAHsCUhRfACsQXiRkAE80aQAiADkvOzT9FXY1DW0AJhYWSSkCRJhoAHJaknq6OggAXlbdctxK6rralvaIUuYgA)

**Comment by erictraut** [comment]
Thanks for the bug report.

There was a bug in the handling of several special cases called out in [PEP 634](https://peps.python.org/pep-0634/#sequence-patterns), which say:

> Although str, bytes, and bytearray are usually considered sequences, they are not included in the above list and do not match sequence patterns.

This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.340, which I just published. It will be included in a future release of pylance.
